packages/utils/src/ast.ts

```diff
@@ -0,0 +1,209 @@
+      // endowment, we can simply remove the argument.
+      //
+      // Note that this only removes `Buffer` from a wrapped function
+      // expression, e.g., `(function (Buffer) { ... })`. Regular functions
+      // are not affected.
+      //
+      // TODO: Since we're working on the AST level, we could check the scope
+      // of the function expression, and possibly prevent false positives?
+      if (node.type === 'FunctionExpression' && node.extra?.parenthesized) {
+        node.params = node.params.filter(
+          (param) => !(param.type === 'Identifier' && param.name === 'Buffer'),
+        );
+      }
+    },
+
+    CallExpression(path) {
+      const { node } = path;
+
+      // Replace `eval(foo)` with `(1, eval)(foo)`.
+      if (node.callee.type === 'Identifier' && node.callee.name === 'eval') {
+        path.replaceWith(
+          evalWrapper({
+            REF: node.callee,
+            ARGS: node.arguments,
+          }) as any,
+        );
+
+        // Need to skip the path here, to prevent the node from being visited
+        // again.
+        path.skip();
+      }
+    },
+
+    MemberExpression(path) {
+      const { node } = path;
+
+      // Replace `object.eval(foo)` with `(1, object.eval)(foo)`.
+      if (
+        node.property.type === 'Identifier' &&
+        node.property.name === 'eval'
+      ) {
+        path.replaceWith(
+          objectEvalWrapper({
+            OBJECT: node.object,
+            REF: node.property,
+          }) as any,
+        );
+
+        // Need to skip the path here, to prevent the node from being visited
+        // again.
+        path.skip();
+      }
+    },
+
+    Identifier(path) {
+      const { node } = path;
+
+      // Insert `regeneratorRuntime` global if it's used in the code.
+      if (node.name === 'regeneratorRuntime') {
+        path.insertBefore(regeneratorRuntimeWrapper());
+        path.stop();
+      }
+    },
+
+    StringLiteral(path) {
+      const { node } = path;
+
+      // Break up tokens that could be parsed as HTML comment terminators, or
+      // `import()` statements.
+      // For reference:
+      // - https://github.com/endojs/endo/blob/70cc86eb400655e922413b99c38818d7b2e79da0/packages/ses/error-codes/SES_HTML_COMMENT_REJECTED.md
+      // - https://github.com/MetaMask/snaps-skunkworks/issues/505
+      //
+      // The RegEx below consists of multiple groups joined by a boolean OR.
+      // Each part consists of two groups which capture a part of each string
+      // which needs to be split up, e.g., `<!--` is split into `<!` and `--`.
+      const tokens = node.value.split(/(<!)(--)|(--)(>)|(import)(\(.*?\))/gu);
+      const replacement = tokens
+        // TODO: The `split` above results in some values being `undefined`.
+        // There may be a better solution to avoid having to filter those out.
+        .filter((value) => value !== '' && value !== undefined)
```

Comment:
This might not be an ideal solution, since it results in `undefined` values for strings containing one of the keywords that need to be split. If anyone has a better suggestion, please let me know.